src/en/azcomic/src/eu/kanade/tachiyomi/extension/en/azcomic/AzComic.kt

```diff
@@ -0,0 +1,341 @@
+        val match = COVER_SLUG_REGEX.find(cover)
+        if (match != null) return match.groupValues[1]
+        val fromUrl = url.substringAfter('/').substringAfter('-')
+        val cleanedUrl = fromUrl.substringBefore("-chapter-")
+        val slug = cleanedUrl.ifBlank { seriesTitle }
+        return slug.toSlug()
+    }
+
+    private fun ComicEntry.chapterNumber(): Float? {
+        val coverMatch = COVER_CHAPTER_REGEX.find(cover)?.groupValues?.get(1)
+        val titleMatch = CHAPTER_NUMBER_REGEX.find(title)?.groupValues?.get(1)
+        val urlMatch = URL_CHAPTER_REGEX.find(url)?.groupValues?.get(1)
+        return listOf(coverMatch, titleMatch, urlMatch)
+            .firstNotNullOfOrNull { it?.toFloatOrNull() }
+    }
+
+    private fun String.toSlug(): String {
+        return lowercase(Locale.ROOT)
+            .replace(Regex("[^a-z0-9]+"), "-")
+            .trim('-')
+    }
+
+    @Serializable
+    private data class ComicEntry(
+        val num: Long,
+        val category: String? = null,
+        val title: String,
+        val cover: String,
+        val url: String,
+        @SerialName("updated_at") val updatedAt: String? = null,
+    )
+
+    @Serializable
+    private data class ImagePayload(
+        val images: List<String>? = null,
+    )
+
+    private data class ChapterEntry(
+        val name: String,
+        val url: String,
+        val dateUpload: Long,
+        val chapterNumber: Float?,
+        val order: Long,
+    )
+
+    private data class SeriesEntry(
+        val slug: String,
+        val title: String,
+        val category: String?,
+        val cover: String?,
+        val latestUrl: String,
+        val updatedAt: Long,
+        val chapters: List<ChapterEntry>,
+    )
+
+    private class CategoryFilter(categories: List<String>) : Filter.Select<String>(
+        "Category",
+        mutableListOf("All").apply { addAll(categories) }.toTypedArray(),
+    ) {
+        private val categoryValues = categories
+        val selected: String? get() = if (state == 0) null else categoryValues[state - 1]
+    }
+
+    private class LetterFilter : Filter.Select<String>(
+        "Alphabetic",
+        LETTER_OPTIONS.toTypedArray(),
+    ) {
+        val selected: String? get() = if (state == 0) null else LETTER_OPTIONS[state]
+    }
+
+    private companion object {
+        private const val PAGE_SIZE = 36
+        private val COVER_SLUG_REGEX = Regex("/uploads/manga/([^/]+)/")
+        private val COVER_CHAPTER_REGEX = Regex("/chapters/([^/]+)/")
+        private val SERIES_TITLE_REGEX = Regex("\\s+(Chapter|Issue)\\b.*$", RegexOption.IGNORE_CASE)
+        private val CHAPTER_NAME_REGEX = Regex("\\b(Chapter|Issue)\\b.*$", RegexOption.IGNORE_CASE)
+        private val CHAPTER_NUMBER_REGEX = Regex("\\b(?:Chapter|Issue)\\s+([0-9]+(?:\\.[0-9]+)?)", RegexOption.IGNORE_CASE)
+        private val URL_CHAPTER_REGEX = Regex("-(?:chapter|issue)-([0-9]+(?:\\.[0-9]+)?)", RegexOption.IGNORE_CASE)
```

Comment:
Put the regex pattern inside strings surrounded by `"""`. Then you can avoid double escaping.